### PR TITLE
Keep project_name shorter than the limit

### LINKED ISSUE #3093: [BUG] Using a long project name causes provisioning failure
Setting a project name of over 64 characters causes provisioning failure.

Traceback:
```
Exception raised during provision.  Incomplete prod_n folder removed.
Traceback (most recent call last):
  File "nvflare/lighter/provision.py", line 189, in <module>
    main()
  File "nvflare/lighter/provision.py", line 185, in main
    handle_provision(args)
  File "nvflare/lighter/provision.py", line 101, in handle_provision
    provision(project_full_path, workspace_full_path, add_user_full_path, add_client_full_path)
  File "nvflare/lighter/provision.py", line 119, in provision
    provisioner.provision(project)
  File "nvflare/lighter/spec.py", line 174, in provision
    b.build(project, ctx)
  File "nvflare/lighter/impl/cert.py", line 123, in build
    self._build_root(project.name, subject_org=None)
  File "nvflare/lighter/impl/cert.py", line 74, in _build_root
    self.root_cert = self._generate_cert(subject, subject_org, self.issuer, pri_key, pub_key, ca=True)
  File "nvflare/lighter/impl/cert.py", line 162, in _generate_cert
    x509_subject = self._x509_name(subject, subject_org, role)
  File "nvflare/lighter/impl/cert.py", line 194, in _x509_name
    name = [x509.NameAttribute(NameOID.COMMON_NAME, cn_name)]
  File "cryptography/x509/name.py", line 152, in __init__
    raise ValueError(msg)
ValueError: Attribute's length must be >= 1 and <= 64, but it was 74
```

**To Reproduce**
1. Create a project config use a name of over 64 characters.
2. Run provisioning.

**Expected behavior**
Successful provisioning, or at least an informative error message.

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04
 - Python Version: 3.10
 - NVFlare Version: 2.5.1

### PR DESCRIPTION
**Pull request by IsaacYangSLA**
Fixes #3093 .

### Description

When project_name in project.yml is longer than 64, generating the root CA cert fails due to CN field limit.  This PR truncates the project_name so it will not be longer than 64.  A friendly message is shown during provisioning if this happens. 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [x] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/lighter/provision.py [hunk 1]
```diff
@@ -142,6 +142,9 @@ def prepare_project(project_dict, add_user_file_path=None, add_client_file_path=
     if api_version not in [3]:
         raise ValueError(f"API version expected 3 but found {api_version}")
     project_name = project_dict.get(PropKey.NAME)
+    if len(project_name) > 63:
+        print(f"Project name {project_name} is longer than 63.  Will truncate it to {project_name[:63]}.")
+        project_name = project_name[:63]
```
[the remark this answers is not in the document]
Comment (IsaacYangSLA) [reply nvflare/lighter/provision.py:147]: Thanks.  The new commit sets the truncated project_name back PropKey.NAME.

**Review by yanchengnv — APPROVED** [review]
LGTM

**Comment by IsaacYangSLA** [comment]
PR was tested with truncated project name.  NVFlare systems were launched without problem.  Job submission worked. 
 Submitted jobs were finished without error.

**Comment by IsaacYangSLA** [comment]
> @IsaacYangSLA do you have the same PR for 2.5 branch ?

Just created the PR for 2.5 branch.  It's a cherry-pick of this PR.

**Pull request merged**